Sources/Glaip/API/Glaip.swift

```diff
@@ -27,35 +27,33 @@ public final class Glaip: ObservableObject {
     self.walletConnect = WalletLinkService(title: title, description: description)
   }
 
-  public func loginUser(type: WalletType, completion: @escaping (Result<String, Error>) -> Void) {
+  public func loginUser(type: WalletType, completion: @escaping (Result<User, Error>) -> Void) {
     switch type {
     case .WalletConnect:
       metaMaskLogin(completion: { [weak self] result in
         guard let self = self else { return }
 
-        DispatchQueue.main.async {
           switch result {
```

Comment:
We actually need to call the main thread for `self.userState = .loggedIn(user)` but I can address it in a separate PR

Reply:
@maurovz Ok I see, I will test that in my case and I will let you know

Reply:
@maurovz I tested it and it still works, thanks bro